src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue29516.cs

```diff
@@ -0,0 +1,23 @@
+﻿#if MACCATALYST || WINDOWS
+// TitleBar is only available on Mac Catalyst and Windows. https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/titlebar?view=net-maui-9.0
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+
+public class Issue29516 : _IssuesUITest
+{
+    public Issue29516(TestDevice device) : base(device) { }
+
+    public override string Issue => "Issue with the TitleBar TrailingContent not being properly aligned on macOS";
+
+    [Test]
+    [Category(UITestCategories.Window)]
+    public void TitleBarTrailingContentShouldRenderProperly()
+    {
+        App.WaitForElement("ContentLabel");
+        VerifyScreenshot();
```

Comment:
Pending snapshots already available in the latest build. Remember, we are clipping the TitleBar from the image by default. You can modify that behavior with parameters.

Reply:
To prevent clipping of the title bar in the images, I have added the necessary parameter in the VerifyScreenshot @jsuarezruiz